overlord/snapstate/policy/errors.go

```diff
@@ -30,6 +30,7 @@ var (
 	errInUseForBoot = errors.New("snap is being used for boot")
 	errRequired     = errors.New("snap is required")
 	errIsModel      = errors.New("snap is used by the model")
+	errNotRemovable = errors.New("snap cannot be removed")
```

Comment:
this is not a good error message: it will result in «snap "snapd" is not removable: snap cannot be removed».
Granted "snap is required" isn't much better (if you don't know what required means in this context).

On core, I'd go maybe with saying it is "essential"; on classic, I'd mention that it needs to be the last thing to go?

Reply:
I updated the code now:
+       errSnapdNotRemovableOnCore       = errors.New("snapd required on core devices")
+       errSnapdNotYetRemovableOnClassic = errors.New("remove all other snaps first")
happy about suggestions of course :)